syntax = "proto3";

package FedLearn;

service FedLearn {
  rpc GetModel (stream Ready) returns (stream ModelReady);
  rpc ModelPoll (Ready) returns (ReadyReply);
}

message Ready {
  string ready = 1;
}

message ReadyReply {
  bool ready = 1;
}

message ModelReady {
  oneof response {
    bool wait = 1;
    int32 model = 2;
  }
}

